core/src/main/java/org/infinispan/interceptors/IsMarshallableInterceptor.java

```diff
@@ -84,6 +85,17 @@ public Object visitGetCacheEntryCommand(InvocationContext ctx, GetCacheEntryComm
    }
 
    @Override
+   public Object visitGetAllCommand(InvocationContext ctx, GetAllCommand command) throws Throwable {
+      if (isStoreAsBinary()) {
+         for (Object key : command.getKeys()) {
+            if (getMightGoRemote(ctx, key, command))
```

Comment:
The marshallable check should happen if storeIsBinary is enabled OR if the command might be replicated to another node. TBH I'm not sure why, the serialization should happen on the user thread in both cases (at least with async marshalling disabled).